docs/build/local-first.md

```diff
@@ -1,25 +1,31 @@
 ---
-sidebar_label: Local-first cache
+sidebar_label: Local-first architecture
 sidebar_position: 3
 description: "If you're building a production-grade app, be sure to use a local-first architecture to help you build a performant app."
 ---
 
 # Use local-first architecture
 
-If you're building a production-grade app, be sure to use a local-first architecture to help you build a performant app. Using this local-first architecture, the client prioritizes using the local cache on the device where it’s running.
+If you're building a production-grade app, be sure to use a local-first architecture to help you build a performant app. Using a local-first architecture means building your app so a client prioritizes using local storage on the device where it’s running.
 
-For example, use the XMTP SDK to initially retrieve existing message data from the XMTP network and place it in the local cache. Asynchronously load new and updated message data as needed. Build your app to get message data from the local cache.
+For example, use the XMTP SDK to initially retrieve existing message data from the XMTP network and place it in local storage. Then, asynchronously load new and updated message data as needed.
 
-**When building a web app with the [React SDK](https://github.com/xmtp/xmtp-web/tree/main/packages/react-sdk), the local-first architecture is automatically provided by the SDK.**
-
-Here’s an overview of how your app frontend, local cache, client SDK, and the XMTP network work together in this local-first approach:
+Here’s an overview of how your app frontend, local storage, client SDK, and the XMTP network work together in this local-first approach:
 
 import localfirst from '/docs/build/img/local-first-arch.jpeg';
 
 <img src={localfirst} style={{width:"450px"}}/>
 
-- When building a web app with the [xmtp-js SDK](https://github.com/xmtp/xmtp-js), you can use the browser `localStorage` as the local cache to store encrypted data, decrypting data each time before display. You might also consider [using Dexie to manage your web app's local data](#manage-local-data-with-dexie-in-a-web-app-built-with-xmtp-js).
+## Mobile apps
+
+- When building a mobile app with the XMTP V3 [React Native](https://github.com/xmtp/xmtp-react-native), [Android](https://github.com/xmtp/xmtp-android), and [iOS](https://github.com/xmtp/xmtp-ios) SDKs, LibXMTP will [create and manage a local database](/docs/concepts/v3/group-chat#local-database-and-syncing) for you. However, if your app supports [reactions](./messages/reaction) and [replies](./messages/reply), you should create and manage a separate local database to store these content types. This local database is key to performantly displaying reactions and replies with their referenced messages when rendering message lists.
```

Comment:
AFAIK, the local storage for reactions and replies only applies to v2 clients. that being said, i believe the mobile SDKs are a v2/v3 hybrid until we migrate everything to v3.

not sure how we can communicate this nuance though. cc @nplasterer

Reply:
Ah, got it - thank you for this clarification, @rygine! Just to riff on this and ensure that I understand - how about this rewrite:

When building a mobile app with the XMTP V3 [React Native](https://github.com/xmtp/xmtp-react-native), [Android](https://github.com/xmtp/xmtp-android), and [iOS](https://github.com/xmtp/xmtp-ios) SDKs, LibXMTP will [create and manage a local database](/docs/concepts/v3/group-chat#local-database-and-syncing) for you.

However, you should create and manage an additional separate local database if your app:
- Supports [reactions](./messages/reaction) and [replies](./messages/reply)
- Uses an XMTP SDK that supports 1:1 messaging only and, more specifically, does not provide the `enableV3` client option to enable group chat.

Use this separate local database to store reactions and replies content types, enabling your app to performantly display them with their referenced messages when rendering message lists.

Reply:
> Uses an XMTP SDK that supports 1:1 messaging only and, more specifically, does not provide the enableV3 client option to enable group chat.

I think if it supports any 1to1 right now it will need a managed database. So maybe

Chooses to support 1:1 to messaging backed by old V2 infrastructure. Or something along those lines. 😬